### PR TITLE
ENH: `Styler.format_index()` to display index values similarly to data-values with `format()`

### PR DESCRIPTION
**Pull request by attack68**
- [x] adds features and method with docstrings
- [x] update user guide
- [x] whats new
- [x] add tests

This PR adds the method `format_index` so that the typical `format` routines can also be applied to display the index labels and columns headers of a Styler.

With all the recent functionality here is a complete example:

Screen Shot 2021-08-18 at 22 55 02

**Notes**

Also had to move `_refactor_levels` function from `style.py` to `style_render.py` for use here.

### CODE REVIEW

**Comment by jreback** [comment]
hmm why do we want to add methods like this? e.g. if you want to string format an index you can already

**Comment by attack68** [comment]
- Styler displays values without altering the underlying DataFrame. Using `pd.Index.format` returns a formatted Index which you *need to overwrite* as the DataFrames Index prior to calling Styler, therefore it breaks the typical usage pattern.
 - `pd.Index.format` does not have the additional options available in `format_index` like HTML/LaTeX escaping, and has a different signature to `Styler.format` which might confuse users.
 - Needing to overwrite the Index with your display format means you can lose functionality of indexing. Here is an example of how this helps preserve functionality:
Screen Shot 2021-08-20 at 08 55 56

 Do throw this out to the team for feedback...

**Comment by attack68** [comment]
**User Guide Updates:**

Screen Shot 2021-08-30 at 20 47 31

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -791,6 +800,175 @@ def format(
...
+        r"""
+        Format the text display value of index labels or column headers.
+
+        .. versionadded:: 1.4.0
+
+        Parameters
+        ----------
+        formatter : str, callable, dict or None
+            Object to define how values are displayed. See notes.
+        axis : int, str
```
Comment (jreback) [inline pandas/io/formats/style_render.py:823]: give the allowed (e.g. 0,1,index,columns)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style_render.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style_render.py]: done

**Review by jreback — APPROVED** [review]

**Comment by attack68** [comment]
these tests consistently fail running the ipython in `source/user_guide/scale.rst` dealing with dask DataFrame. 
But this PR is unrelated to that so I don't know the reason. Couldn't really find other PRs suffering from the same test failures?

**Comment by jreback** [comment]
> these tests consistently fail running the ipython in `source/user_guide/scale.rst` dealing with dask DataFrame.
> But this PR is unrelated to that so I don't know the reason. Couldn't really find other PRs suffering from the same test failures?
> The docs build locally without errors so its puzzling.

hmm, did dask version change?

**Comment by jreback** [comment]
looks like some actual doc failures: https://github.com/pandas-dev/pandas/pull/43101/checks?check_run_id=3570623205

**Comment by attack68** [comment]
> looks like some actual doc failures: https://github.com/pandas-dev/pandas/pull/43101/checks?check_run_id=3570623205

everytime says `build completed with 8 warnings`, and every warning is in respect to `/home/runner/work/pandas/pandas/doc/source/user_guide/scale.rst`, which is an unaltered file, and never calls Styler or the impacted files here.

The docs build locally without warning so I am puzzled, will see if I can come up with anything

**Comment by attack68** [comment]
@jreback managed to get this to pass after merge. revert if necessary but I think should be ok now.

**Comment by jreback** [comment]
ok great. very odd that that was happening,

**Pull request merged**